ivy/functional/ivy/linear_algebra.py

```diff
@@ -380,26 +383,173 @@ def diagonal(
         last two dimensions and appending a dimension equal to the size of the resulting
         diagonals. The returned array must have the same data type as ``x``.
 
-    Examples
-    --------
+    Functional Examples
+    ------------------
+
+    With :code:`ivy.Array` inputs:
+    
     >>> x = ivy.array([[1., 2.],\
                        [3., 4.]])
     >>> d = ivy.diagonal(x)
     >>> print(d)
     ivy.array([1., 4.])
+
+    >>> x = ivy.array([[1., 2.],\
+                       [3., 4.]])
     >>> d = ivy.diagonal(x, 1)
     >>> print(d)
     ivy.array([1.])
 
-    A 3-D Example
     >>> x = ivy.array([[[1., 2.],\
                         [3., 4.]],\
                        [[5., 6.],\
                         [7., 8.]]])
-    >>> d = ivy.diagonal(x, 0, 0, 1)
+    >>> d = ivy.diagonal(x)
     >>> print(d)
-    ivy.array([[1., 7.],
-               [2., 8.]])
+    ivy.array([[1., 4.],
+               [5., 8.]])
+
+    >>> y = ivy.arange(9).reshape((3, 3))
+    >>>print(y)
+    ivy.array([[0, 1, 2],
+               [3, 4, 5],
+               [6, 7, 8]], dtype=int32)
+
+    >>> d = ivy.diagonal(y, -1)
```

Comment:
And for some of these you need to specify again the full array `y` that is being used at the start of the example.

Please go through the examples again to make changes where necessary and let me know when you have made the edits!